CMakeLists.txt

```diff
@@ -43,20 +43,33 @@ elseif(APPLE)
     set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 endif()
 
-function(add_cxx_flag_if_supported flag)
+function(check_if_cxx_compiler_flag_supported flag out_var)
     if(flag MATCHES "^-Wno-")
         # Compilers ignore unknown -Wno-foo flags, so look for -Wfoo instead.
         string(REPLACE "-Wno-" "-W" check_flag ${flag})
     else()
         set(check_flag ${flag})
     endif()
 
-    check_cxx_compiler_flag(${check_flag} HAVE${check_flag})
-    if(HAVE${check_flag})
+    check_cxx_compiler_flag(${check_flag} ${out_var})
+endfunction()
+
+function(add_cxx_flag_if_supported flag)
+    check_if_cxx_compiler_flag_supported(${flag} HAVE${flag})
+    if(HAVE${flag})
         add_compile_options($<$<COMPILE_LANGUAGE:CXX>:${flag}>)
     endif()
 endfunction()
 
+function(append_source_file_compile_options file property)
```

Comment:
that's a bit defeats the purpose of using set_source_files_properties. It's for multiple files.
Also better to use 'target_compile_options' - it's more general (acts not on just files), and also supports options deduplication automatically.